<a href="https://colab.research.google.com/github/fermuba/ML_OPS_MOVIES/blob/main/ETL/DS_Movies.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ingesta de Data set Proyecto Individual HENRY DT-PT01 movies_dataset.csv

In [1]:
# Acceso a los archivos de Drive
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# Carga de Librerias de uso
import pandas as pd
import numpy as np
import json
import ast # Abstract Syntax Trees

# configuro semilla de random
np.random.seed(0)

In [3]:
# Insgesta del archivos movie_dataset.csv
movie = pd.read_csv('/content/drive/MyDrive/Colab/HENRY/PI-1/Data_font/movies_dataset.csv')

<ipython-input-3-e643f2711934>:2: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  movie = pd.read_csv('/content/drive/MyDrive/Colab/HENRY/PI-1/Data_font/movies_dataset.csv')


Tomamos contacto con el dataset y realizamos una primera revision general para ver cual es el estado y tener conocimiento de que tareas hay que realizar

In [4]:
# Informacion
movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

In [5]:
# Encabezado parcial
movie.loc[:, : 'poster_path'].head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                        poster_path  
0  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg  
1  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg  
2  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg  
3  /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg  
4  /e64sOI48hQXyru7naBFyssKFxVd.jpg

In [6]:
# Encabezado parcial
movie.loc[:, 'production_companies' : ].head()

production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video  vote_average  vote_count  
0                    Toy Story  False           7.7      5415.0  
1                      Jumanji  False           6.9      2413.0  
2             Grumpier Old Men  False           6.5        92.0  
3            Waiting to Exhale  False           6.1        34.0  
4  Father of the Bride Part II  False           5.7       173.0

El dataset tiene un tamaño de 45466 filas x 24 columns.
Lo primero que voy a hacer es tomar el diccionario y crear un par de columnas para tener una guia de tareas a ejecutar con un seguimiento para  identificar la cantidad de operaciones a realizar.
A continuacion voy a listar las tareas para este dataframe.
- Eliminacion de columnas solicitadas
- Reemplazar valores NaN y vacios
- Desanidar columnas
- Eliminar columnas generadas por el paso anterior que no se necesiten



In [7]:
# Eliminacion de columnas
columnas_eliminar = ['video', 'imdb_id', 'adult', 'original_title', 'poster_path', 'homepage']
movie = movie.drop(columns= columnas_eliminar)

movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   belongs_to_collection  4494 non-null   object 
 1   budget                 45466 non-null  object 
 2   genres                 45466 non-null  object 
 3   id                     45466 non-null  object 
 4   original_language      45455 non-null  object 
 5   overview               44512 non-null  object 
 6   popularity             45461 non-null  object 
 7   production_companies   45463 non-null  object 
 8   production_countries   45463 non-null  object 
 9   release_date           45379 non-null  object 
 10  revenue                45460 non-null  float64
 11  runtime                45203 non-null  float64
 12  spoken_languages       45460 non-null  object 
 13  status                 45379 non-null  object 
 14  tagline                20412 non-null  object 
 15  ti

In [8]:
# Reemplazo el nombre de la columna id
movie.rename(columns={'id': 'id_movie'}, inplace= True)

In [9]:
# Tamaño Filas x Columnas original
movie.shape

(45466, 18)

In [10]:
# Eliminacion de filas duplicadas comparando ['id_movie', 'title']
indice = movie[movie.duplicated(subset= ['id_movie', 'title'], keep= 'first')].index
movie.drop(index= indice, inplace= True)

In [11]:
# Nuevo tamaño
movie.shape

(45436, 18)

In [12]:
# Saco la columna genre a otro dataframe
genero = movie.loc[:, ['id_movie', 'genres']]
movie = movie.drop(columns= 'genres')
genero.to_csv('/content/drive/MyDrive/Colab/HENRY/PI-1/Data_font/movies_genres.csv')


Desanidar la columna 'belongs_to_collection'

In [13]:
movie['belongs_to_collection']

0        {'id': 10194, 'name': 'Toy Story Collection', ...
1                                                      NaN
2        {'id': 119050, 'name': 'Grumpy Old Men Collect...
3                                                      NaN
4        {'id': 96871, 'name': 'Father of the Bride Col...
                               ...                        
45461                                                  NaN
45462                                                  NaN
45463                                                  NaN
45464                                                  NaN
45465                                                  NaN
Name: belongs_to_collection, Length: 45436, dtype: object

In [14]:
# Paso los datos str a datos interpretados por python
selection_data = lambda column_data: ast.literal_eval(column_data) if pd.notna(column_data) else column_data
movie['belongs_to_collection'] = movie['belongs_to_collection'].apply(selection_data)

In [15]:
# Contenido de un elemento de la columna seleccionada en la primera fila
movie['belongs_to_collection'][0]

{'id': 10194,
 'name': 'Toy Story Collection',
 'poster_path': '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg',
 'backdrop_path': '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'}

In [16]:
# Expansion de los datos del json en columnas
columna_expandida = pd.json_normalize(movie['belongs_to_collection'])

In [17]:
# Informacion de la columna nueva
columna_expandida.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45436 entries, 0 to 45435
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             4488 non-null   float64
 1   name           4488 non-null   object 
 2   poster_path    3945 non-null   object 
 3   backdrop_path  3261 non-null   object 
dtypes: float64(1), object(3)
memory usage: 1.4+ MB


In [18]:
# Eliminar columnas innecesarias
columnas_eliminar = ['id', 'poster_path', 'backdrop_path']
columna_expandida.drop(columns= columnas_eliminar, inplace= True)

In [19]:
# Remplazo la informacion corregida y completo los valores NaN
movie['belongs_to_collection'] = columna_expandida
movie['belongs_to_collection'].fillna('Sin Información', inplace= True)


## Limpieza de production_companies

In [20]:
production_companies = movie['production_companies']

In [21]:
production_companies.head()

0       [{'name': 'Pixar Animation Studios', 'id': 3}]
1    [{'name': 'TriStar Pictures', 'id': 559}, {'na...
2    [{'name': 'Warner Bros.', 'id': 6194}, {'name'...
3    [{'name': 'Twentieth Century Fox Film Corporat...
4    [{'name': 'Sandollar Productions', 'id': 5842}...
Name: production_companies, dtype: object

In [22]:
production_companies = production_companies.apply(selection_data)

In [23]:
columna_expandida = pd.json_normalize(production_companies)
production_companies = pd.concat([production_companies, columna_expandida], axis= 1)

In [24]:
production_companies.head(3)

production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   

                                              0  \
0  {'name': 'Pixar Animation Studios', 'id': 3}   
1       {'name': 'TriStar Pictures', 'id': 559}   
2          {'name': 'Warner Bros.', 'id': 6194}   

                                         1  \
0                                     None   
1     {'name': 'Teitler Film', 'id': 2550}   
2  {'name': 'Lancaster Gate', 'id': 19464}   

                                                   2     3     4     5     6  \
0                                               None  None  None  None  None   
1  {'name': 'Interscope Communications', 'id': 10...  None  None  None  None   
2                                               None  None  None  None  None   

      7     8  ...    16    17    18    19    20    21    22    23    24    25  
0  None  None  ...  None  None  None  None  None  None  None  None  None  None  
1  None  None  ...  None  None  None  None  None  None  None  None  None  None  
2  None  None  ...  None  None  None  None  None  None  None  None  None  None  

[3 rows x 27 columns]

In [25]:
production_companies.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45466 entries, 0 to 45265
Data columns (total 27 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   production_companies  45433 non-null  object
 1   0                     33562 non-null  object
 2   1                     16992 non-null  object
 3   2                     9032 non-null   object
 4   3                     4321 non-null   object
 5   4                     2374 non-null   object
 6   5                     1426 non-null   object
 7   6                     865 non-null    object
 8   7                     552 non-null    object
 9   8                     373 non-null    object
 10  9                     240 non-null    object
 11  10                    173 non-null    object
 12  11                    131 non-null    object
 13  12                    97 non-null     object
 14  13                    78 non-null     object
 15  14                    62 non-null   

In [26]:
for i in range(0, 26):
  columna_expandida = pd.json_normalize(production_companies[i])
  production_companies = pd.concat([production_companies, columna_expandida], axis= 1)

In [27]:
production_companies.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45466 entries, 0 to 45265
Data columns (total 79 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   production_companies  45433 non-null  object 
 1   0                     33562 non-null  object 
 2   1                     16992 non-null  object 
 3   2                     9032 non-null   object 
 4   3                     4321 non-null   object 
 5   4                     2374 non-null   object 
 6   5                     1426 non-null   object 
 7   6                     865 non-null    object 
 8   7                     552 non-null    object 
 9   8                     373 non-null    object 
 10  9                     240 non-null    object 
 11  10                    173 non-null    object 
 12  11                    131 non-null    object 
 13  12                    97 non-null     object 
 14  13                    78 non-null     object 
 15  14                 

In [28]:
# Eliminacion de columnas
columnas_eliminar = ['production_companies', 'id']
production_companies.drop(columns= columnas_eliminar, axis= 1, inplace= True)
columnas_eliminar = np.arange(26)
production_companies.drop(columns= columnas_eliminar, axis= 1, inplace= True)


In [29]:
production_companies.head()

name                 name  \
0                 Pixar Animation Studios                  NaN   
1                        TriStar Pictures         Teitler Film   
2                            Warner Bros.       Lancaster Gate   
3  Twentieth Century Fox Film Corporation                  NaN   
4                   Sandollar Productions  Touchstone Pictures   

                        name name name name name name name name  ... name  \
0                        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN   
1  Interscope Communications  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN   
2                        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN   
3                        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN   
4                        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN   

  name name name name name name name name name  
0  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
1  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
2  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
3  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
4  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[5 rows x 26 columns]

In [30]:
nombre_columnas = ['col1', 'col2', 'col3', 'col4', 'col5', 'col6',
                   'col7', 'col8', 'col9', 'col10', 'col11', 'col12',
                   'col13', 'col14', 'col15', 'col16', 'col17', 'col18',
                   'col19', 'col20', 'col21', 'col22', 'col23', 'col24',
                   'col25', 'col26']

production_companies.columns = nombre_columnas

In [31]:
production_companies.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45466 entries, 0 to 45265
Data columns (total 26 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   col1    33562 non-null  object
 1   col2    16992 non-null  object
 2   col3    9032 non-null   object
 3   col4    4321 non-null   object
 4   col5    2374 non-null   object
 5   col6    1426 non-null   object
 6   col7    865 non-null    object
 7   col8    552 non-null    object
 8   col9    373 non-null    object
 9   col10   240 non-null    object
 10  col11   173 non-null    object
 11  col12   131 non-null    object
 12  col13   97 non-null     object
 13  col14   78 non-null     object
 14  col15   62 non-null     object
 15  col16   53 non-null     object
 16  col17   32 non-null     object
 17  col18   25 non-null     object
 18  col19   22 non-null     object
 19  col20   17 non-null     object
 20  col21   12 non-null     object
 21  col22   8 non-null      object
 22  col23   5 non-null    

In [32]:
# Eliminacion de los elementos NaN por fila y creacion de una nueva columna con la lista de elementos
production_companies['production_companies'] = production_companies.apply(lambda fila: fila.dropna().tolist(), axis=1)

In [33]:
type(production_companies.production_companies)


pandas.core.series.Series

In [34]:
production_companies.head()

col1                 col2  \
0                 Pixar Animation Studios                  NaN   
1                        TriStar Pictures         Teitler Film   
2                            Warner Bros.       Lancaster Gate   
3  Twentieth Century Fox Film Corporation                  NaN   
4                   Sandollar Productions  Touchstone Pictures   

                        col3 col4 col5 col6 col7 col8 col9 col10  ... col18  \
0                        NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  ...   NaN   
1  Interscope Communications  NaN  NaN  NaN  NaN  NaN  NaN   NaN  ...   NaN   
2                        NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  ...   NaN   
3                        NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  ...   NaN   
4                        NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN  ...   NaN   

  col19 col20 col21 col22 col23 col24 col25 col26  \
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   

                                production_companies  
0                          [Pixar Animation Studios]  
1  [TriStar Pictures, Teitler Film, Interscope Co...  
2                     [Warner Bros., Lancaster Gate]  
3           [Twentieth Century Fox Film Corporation]  
4       [Sandollar Productions, Touchstone Pictures]  

[5 rows x 27 columns]

In [35]:
# Reemplazo los datos de la columna original por los datos limpios
movie['production_companies'] = production_companies['production_companies']

movie.head()

belongs_to_collection    budget id_movie original_language  \
0            Toy Story Collection  30000000      862                en   
1                 Sin Información  65000000     8844                en   
2       Grumpy Old Men Collection         0    15602                en   
3                 Sin Información  16000000    31357                en   
4  Father of the Bride Collection         0    11862                en   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                                production_companies  \
0                          [Pixar Animation Studios]   
1  [TriStar Pictures, Teitler Film, Interscope Co...   
2                     [Warner Bros., Lancaster Gate]   
3           [Twentieth Century Fox Film Corporation]   
4       [Sandollar Productions, Touchstone Pictures]   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  vote_average  vote_count  
0                    Toy Story           7.7      5415.0  
1                      Jumanji           6.9      2413.0  
2             Grumpier Old Men           6.5        92.0  
3            Waiting to Exhale           6.1        34.0  
4  Father of the Bride Part II           5.7       173.0

## Limpieza de production_countries
Los convierto de str a json y quedan de esa forma, no se van a usar

In [36]:
movie['production_countries'] = movie['production_countries'].apply(selection_data)


In [37]:
movie.production_countries.fillna('Sin Información', inplace= True)

## Limpieza de spoken_languages
Los convierto de str a json y quedan de esa forma, no se van a usar

In [38]:
movie['spoken_languages'] = movie['spoken_languages'].apply(selection_data)

In [39]:
movie.spoken_languages.fillna('Sin Información', inplace= True)

In [40]:
movie['spoken_languages'][0][0]

{'iso_639_1': 'en', 'name': 'English'}

### Operaciones adicionales solicitadas

In [42]:
# Revision de budget
movie['budget'] = pd.to_numeric(movie['budget'], errors= 'coerce')
print("El tamaño es: ", movie[movie.budget.isna()].shape)
movie.budget.fillna(0, inplace= True)
print("El tipo de archivo es ", type(movie.budget))
print("El Minimo de la columna es: ", movie.budget.min())
print("El Maximo de la columna es: ", movie.budget.max())

El tamaño es:  (0, 17)
El tipo de archivo es  <class 'pandas.core.series.Series'>
El Minimo de la columna es:  0.0
El Maximo de la columna es:  380000000.0


In [ ]:
# Cantidad de nulos
movie[movie.revenue.isnull()].shape

(6, 17)

In [43]:
# Revision de revenue
movie['revenue'] = pd.to_numeric(movie['revenue'], errors= 'coerce')
print("El tamaño es: ", movie[movie.revenue.isna()].shape)
movie.revenue.fillna(0, inplace= True)
print("El tipo de archivo es ", type(movie.revenue))
print("El Minimo de la columna es: ", movie.revenue.min())
print("El Maximo de la columna es: ", movie.revenue.max())

El tamaño es:  (6, 17)
El tipo de archivo es  <class 'pandas.core.series.Series'>
El Minimo de la columna es:  0.0
El Maximo de la columna es:  2787965087.0


In [ ]:
# Revision de release_date
movie['release_date'] = pd.to_datetime(movie['release_date'], errors= 'coerce')
movie['release_date'] = movie['release_date'].dt.date
print(movie[movie.release_date.isna()].shape)
movie.release_date.fillna(0, inplace= True)

(90, 17)
<class 'pandas.core.series.Series'>


In [ ]:
# Cantidad de elementos 0 en la columna release_date
movie[movie.release_date == 0].shape

(90, 17)

### Completado de nulos del resto de columnas

In [ ]:
print(movie[movie.id_movie.isna()].shape)
print(movie[movie.original_language.isna()].shape)
print(movie[movie.overview.isna()].shape)
print(movie[movie.popularity.isna()].shape)
print(movie[movie.runtime.isna()].shape)
print(movie[movie.status.isna()].shape)
print(movie[movie.tagline.isna()].shape)
print(movie[movie.title.isna()].shape)
print(movie[movie.vote_average.isna()].shape)
print(movie[movie.vote_count.isna()].shape)

(0, 17)
(11, 17)
(954, 17)
(5, 17)
(263, 17)
(87, 17)
(25035, 17)
(6, 17)
(6, 17)
(6, 17)


In [ ]:
movie.original_language.fillna('Sin Informacion', inplace= True)
movie.overview.fillna('Sin Informacion', inplace= True)
movie.popularity.fillna('Sin Informacion', inplace= True)
movie.runtime.fillna('Sin Informacion', inplace= True)
movie.status.fillna('Sin Informacion', inplace= True)
movie.tagline.fillna('Sin Informacion', inplace= True)
movie.title.fillna('Sin Informacion', inplace= True)
movie.vote_average.fillna('Sin Informacion', inplace= True)
movie.vote_count.fillna('Sin Informacion', inplace= True)

In [ ]:
print(movie[movie.id_movie.isna()].shape)
print(movie[movie.original_language.isna()].shape)
print(movie[movie.overview.isna()].shape)
print(movie[movie.popularity.isna()].shape)
print(movie[movie.runtime.isna()].shape)
print(movie[movie.status.isna()].shape)
print(movie[movie.tagline.isna()].shape)
print(movie[movie.title.isna()].shape)
print(movie[movie.vote_average.isna()].shape)
print(movie[movie.vote_count.isna()].shape)

(0, 17)
(0, 17)
(0, 17)
(0, 17)
(0, 17)
(0, 17)
(0, 17)
(0, 17)
(0, 17)
(0, 17)


## Creacion de Columnas nuevas

In [ ]:
# Creacion de release_year
movie['release_year'] = pd.to_datetime(movie.loc[movie['release_date'] != 0, 'release_date']).dt.year
movie.release_year.fillna(0, inplace= True)
movie['release_year'] = movie['release_year'].astype(int)
movie.loc[: ,['id_movie', 'release_date', 'release_year']]

id_movie release_date  release_year
0          862   1995-10-30          1995
1         8844   1995-12-15          1995
2        15602   1995-12-22          1995
3        31357   1995-12-22          1995
4        11862   1995-02-10          1995
...        ...          ...           ...
45461   439050            0             0
45462   111109   2011-11-17          2011
45463    67758   2003-08-01          2003
45464   227506   1917-10-21          1917
45465   461257   2017-06-09          2017

[45436 rows x 3 columns]

In [ ]:
# Creacion de columna return

movie['return'] = movie['revenue'] / movie.budget[movie['budget'] != 0]
movie['return'].fillna(0, inplace= True)

In [ ]:
# Exportacion de la informacion procesada a un archivo
movie.to_csv('/content/drive/MyDrive/Colab/HENRY/PI-1/Data_font/movies_clean.csv')